awx/main/utils/common.py

```diff
@@ -329,12 +329,19 @@ def update_scm_url(scm_type, url, username=True, password=True, check_special_ca
 
     # Special handling for github/bitbucket SSH URLs.
     if check_special_cases:
-        special_git_hosts = ('github.com', 'bitbucket.org', 'altssh.bitbucket.org')
-        if scm_type == 'git' and parts.scheme.endswith('ssh') and parts.hostname in special_git_hosts and netloc_username != 'git':
-            raise ValueError(_('Username must be "git" for SSH access to %s.') % parts.hostname)
-        if scm_type == 'git' and parts.scheme.endswith('ssh') and parts.hostname in special_git_hosts and netloc_password:
-            # raise ValueError('Password not allowed for SSH access to %s.' % parts.hostname)
-            netloc_password = ''
+        special_hosts = ('github.com', 'bitbucket.org', 'altssh.bitbucket.org')
+        allowed_git_usernames = {'git', 'x-access-token'}
+
+        if scm_type == 'git' and parts.scheme.endswith('ssh'):
+            is_github_host = parts.hostname in special_hosts or parts.hostname.endswith('.github.com')
```

Comment:
seems like we could have just done

```python
if scm_type == 'git' and parts.scheme.endswith('ssh') and parts.hostname in special_git_hosts and not netloc_username in allowed_git_username:
```

Reply:
and followup question, does this x-access-token work for bit bucket? if not, do we need to add validation for that case?